File: mobile/lib/models/booking.dart
import 'enums.dart';

class BookingInfo {
  final String id;
  final String careRequestId;
  final String applicationId;
  final String patientId;
  final String nurseId;
  final BookingStatus status;

  final DateTime startDate;
  final DateTime? endDate;
  final double? hoursPerDay;
  final PriceUnit paymentFrequency;
  final double? agreedPrice;

  const BookingInfo({
    required this.id,
    required this.careRequestId,
    required this.applicationId,
    required this.patientId,
    required this.nurseId,
    required this.status,
    required this.startDate,
    this.endDate,
    this.hoursPerDay,
    required this.paymentFrequency,
    this.agreedPrice,
  });

  factory BookingInfo.fromJson(Map<String, dynamic> j) => BookingInfo(
        id: j['id'] as String,
        careRequestId: j['care_request_id'] as String,
        applicationId: j['application_id'] as String,
        patientId: j['patient_id'] as String,
        nurseId: j['nurse_id'] as String,
        status: bookingStatusFromApi(j['status'] as String),
        startDate: DateTime.parse(j['start_date'] as String),
        endDate: j['end_date'] != null ? DateTime.parse(j['end_date'] as String) : null,
        hoursPerDay: (j['hours_per_day'] as num?)?.toDouble(),
        paymentFrequency: priceUnitFromApi(j['payment_frequency'] as String),
        agreedPrice: (j['agreed_price'] as num?)?.toDouble(),
      );
}

File: backend/app/schemas/booking.py
import uuid
from datetime import date

from pydantic import BaseModel

from app.models.booking import BookingStatus
from app.models.nurse import PriceUnit


class BookingResponse(BaseModel):
    id: uuid.UUID
    care_request_id: uuid.UUID
    application_id: uuid.UUID
    patient_id: uuid.UUID
    nurse_id: uuid.UUID
    status: BookingStatus

    start_date: date
    end_date: date | None
    hours_per_day: float | None
    payment_frequency: PriceUnit
    agreed_price: float | None

    model_config = {"from_attributes": True}


def booking_to_response(booking) -> "BookingResponse":
    return BookingResponse(
        id=booking.id,
        care_request_id=booking.care_request_id,
        application_id=booking.application_id,
        patient_id=booking.patient_id,
        nurse_id=booking.nurse_id,
        status=booking.status,
        start_date=booking.start_date,
        end_date=booking.end_date,
        hours_per_day=float(booking.hours_per_day) if booking.hours_per_day is not None else None,
        payment_frequency=booking.payment_frequency,
        agreed_price=float(booking.agreed_price) if booking.agreed_price is not None else None,
    )
